homeassistant/helpers/trigger.py

```diff
@@ -409,3 +514,108 @@ def remove_triggers() -> None:
             remove()
 
     return remove_triggers
+
+
+def _load_triggers_file(hass: HomeAssistant, integration: Integration) -> JSON_TYPE:
+    """Load triggers file for an integration."""
+    try:
+        return cast(
+            JSON_TYPE,
+            _TRIGGERS_SCHEMA(
+                load_yaml_dict(str(integration.file_path / "triggers.yaml"))
+            ),
+        )
+    except FileNotFoundError:
+        _LOGGER.warning(
+            "Unable to find triggers.yaml for the %s integration", integration.domain
+        )
+        return {}
+    except (HomeAssistantError, vol.Invalid) as ex:
+        _LOGGER.warning(
+            "Unable to parse triggers.yaml for the %s integration: %s",
+            integration.domain,
+            ex,
+        )
+        return {}
+
+
+def _load_triggers_files(
+    hass: HomeAssistant, integrations: Iterable[Integration]
+) -> dict[str, JSON_TYPE]:
+    """Load trigger files for multiple integrations."""
+    return {
+        integration.domain: _load_triggers_file(hass, integration)
+        for integration in integrations
+    }
+
+
+async def async_get_all_descriptions(
+    hass: HomeAssistant,
+) -> dict[str, dict[str, Any]]:
+    """Return descriptions (i.e. user documentation) for all triggers."""
+    descriptions_cache = hass.data[TRIGGER_DESCRIPTION_CACHE]
+
+    triggers = hass.data[TRIGGERS]
+    # See if there are new triggers not seen before.
+    # Any trigger that we saw before already has an entry in description_cache.
+    all_triggers = set(triggers)
+    # If we have a complete cache, check if it is still valid
+    if all_cache := hass.data.get(ALL_TRIGGER_DESCRIPTIONS_CACHE):
+        previous_all_triggers = set(all_cache)
+        # If the triggers are the same, we can return the cache
+        if previous_all_triggers == all_triggers:
+            return all_cache
+
+    # Files we loaded for missing descriptions
+    loaded: dict[str, JSON_TYPE] = {}
```

Comment:
```suggestion
    new_triggers_descriptions: dict[str, JSON_TYPE] = {}
```
or `loaded_files_for_missing_triggers`, `missing_triggers_loaded_files`, or just `missing_triggers_files`, or something like that to make it clearer (and avoid needing a comment). but I'm ok with also keeping the comment in addition to renaming

Reply:
Fixed